sim/druid/items.go

```diff
@@ -107,6 +108,72 @@ var ItemSetBattlegearOfTheEternalBlossom = core.NewItemSet(core.ItemSet{
 	},
 })
 
+// Feral PvP
+var ItemSetGladiatorSanctuary = core.NewItemSet(core.ItemSet{
+	Name:                    "Gladiator's Sanctuary",
+	DisabledInChallengeMode: true,
+	Bonuses: map[int32]core.ApplySetBonus{
+		2: func(_ core.Agent, _ *core.Aura) {
+			// Not implemented
+		},
+		4: func(agent core.Agent, setBonusAura *core.Aura) {
+			// Once every 30 sec, your next Ravage is free and has no positional or stealth requirement.
+			druid := agent.(DruidAgent).GetDruid()
+			if druid.Spec == proto.Spec_SpecFeralDruid {
+				druid.registerStampede()
+				druid.registerStampedePending()
+				setBonusAura.ApplyOnEncounterStart(func(_ *core.Aura, sim *core.Simulation) {
+					druid.StampedeAura.Activate(sim)
+				})
+			}
+		},
+	},
+})
+
+func (druid *Druid) registerStampede() {
+	var oldExtraCastCondition core.CanCastCondition
+
+	druid.StampedeAura = druid.RegisterAura(core.Aura{
+		Label:    "Stampede",
+		ActionID: core.ActionID{SpellID: 81022},
+		Duration: core.NeverExpires,
+
+		OnGain: func(_ *core.Aura, _ *core.Simulation) {
+			oldExtraCastCondition = druid.Ravage.ExtraCastCondition
+			druid.Ravage.ExtraCastCondition = nil
+			druid.Ravage.Cost.FlatModifier -= 45
+		},
+
+		OnSpellHitDealt: func(_ *core.Aura, sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			if spell.Matches(DruidSpellRavage) && result.Landed() {
+				druid.StampedeAura.Deactivate(sim)
+				druid.StampedePendingAura.Activate(sim)
+			}
+		},
+
+		OnExpire: func(_ *core.Aura, _ *core.Simulation) {
+			druid.Ravage.ExtraCastCondition = oldExtraCastCondition
+			druid.Ravage.Cost.FlatModifier += 45
+		},
+
+		OnEncounterStart: func(_ *core.Aura, sim *core.Simulation) {
+			druid.StampedeAura.Activate(sim)
+		},
```

Comment:
Remove this, because you're already triggering an activation on encounter start if the set bonus is active at that time.